src/hotspot/share/runtime/globals.hpp

```diff
@@ -680,6 +680,10 @@ const int ObjectAlignmentInBytes = 8;
           "Allow parallel defineClass requests for class loaders "          \
           "registering as parallel capable")                                \
                                                                             \
+  product(bool, SynchronizeLoadClass, false, DIAGNOSTIC,                    \
```

Comment:
I don't think the flag can be diagnostic. If people need to revert behaviour for their production systems then they will want a product level flag to use. This will need a CSR request anyway.

Reply:
As diagnostic, it is a product level flag.